pkg/nfd-master/node-updater-pool.go

```diff
@@ -0,0 +1,86 @@
+package nfdmaster
+
+import (
+	"sync"
+
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+type nodeUpdaterPool struct {
+	queue     workqueue.RateLimitingInterface
+	wg        sync.WaitGroup
+	nfdMaster *nfdMaster
+	started   bool
+	mutex     sync.Mutex
+}
+
+func newNodeUpdaterPool(nfdMaster *nfdMaster) *nodeUpdaterPool {
+	return &nodeUpdaterPool{
+		nfdMaster: nfdMaster,
+		wg:        sync.WaitGroup{},
+		mutex:     sync.Mutex{},
+	}
+}
+
+func (u *nodeUpdaterPool) processNodeLabelRequest(queue workqueue.RateLimitingInterface) bool {
+	nodeName, quit := queue.Get()
+	if quit {
+		return false
+	}
+
+	defer queue.Done(nodeName)
+
+	if err := u.nfdMaster.nfdAPIUpdateOneNode(nodeName.(string)); err != nil {
+		if queue.NumRequeues(nodeName) < 5 {
+			klog.InfoS("retrying labeling request for node", "nodeName", nodeName)
+			queue.AddRateLimited(nodeName)
+			return true
+		} else {
+			klog.ErrorS(err, "error labeling node", "nodeName", nodeName)
+		}
+	}
+	queue.Forget(nodeName)
+	return true
+}
+
+func (u *nodeUpdaterPool) runNodeUpdater(queue workqueue.RateLimitingInterface) {
+	for u.processNodeLabelRequest(queue) {
+	}
+	u.wg.Done()
+}
+
+func (u *nodeUpdaterPool) start(parallelism int) {
+	u.mutex.Lock()
+	defer u.mutex.Unlock()
+
+	if u.started {
```

Comment:
We could do away with the started bool with smth like

```suggestion
	if u.queue != nil && !u.queue.ShuttingDown() {
```